### PR TITLE
Reply functionality for Group Chat and DirectChat

### LINKED ISSUE #2103: Add reply to messages functionality in direct chat and group chat (GSoC)
**Is your feature request related to a problem? Please describe.**
Add reply to messages functionality in direct chat and group chat
**Describe the solution you'd like**
Users should be able to reply to messages in direct chat and group chat
**Describe alternatives you've considered**

**Approach to be followed (optional)**
A clear and concise description of approach to be followed.

**Additional context**
image

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by disha1202**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #2103 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

- **New Features**
	- Introduced new styles for the chat room interface, enhancing layout and user interaction.
	- Improved hover effects and animations for sent and received messages, increasing visual feedback.
	- Enhanced loading state in the user modal for adding existing users, improving user experience.

- **Bug Fixes**
	- Resolved issues related to the organization selection in the group chat creation process, preventing runtime errors.

- **Style**
	- Updated CSS for the chat room to improve layout, including the addition of a new user details class and enhanced message styles.

- **Tests**
	- Expanded test cases for the ChatRoom component, ensuring accurate simulation of user interactions and message handling scenarios.
	- Simplified test descriptions for clarity and maintainability.

### CODE REVIEW

File: src/GraphQl/Mutations/OrganizationMutations.ts [hunk 1]
```diff
@@ -57,15 +57,174 @@ export const REMOVE_SAMPLE_ORGANIZATION_MUTATION = gql`
...
+      }
+      sender {
+        _id
+        firstName
+        lastName
+      }
+      updatedAt
+    }
+  }
+`;
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/GraphQl/Mutations/OrganizationMutations.ts]: No chatId is not required here

File: src/GraphQl/Mutations/OrganizationMutations.ts [hunk 2]
```diff
@@ -57,15 +57,174 @@ export const REMOVE_SAMPLE_ORGANIZATION_MUTATION = gql`
...
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/GraphQl/Mutations/OrganizationMutations.ts]: not required

**Comment by palisadoes** [comment]
@DMills27 @Kevoniat PTAL

**Comment by palisadoes** [comment]
Please get the code coverage on the patch higher

**Comment by palisadoes** [comment]
@disha1202 Please increase the code coverage of the patch

**Comment by palisadoes** [comment]
@disha1202 Please work with your mentor to merge your PRs in the correct order after they approve your work

**Comment by disha1202** [comment]
> @disha1202 Please work with your mentor to merge your PRs in the correct order after they approve your work

Okay @palisadoes

**Comment by palisadoes** [comment]
Please work on getting the code-rabbit tests to pass too.

File: src/components/UserPortal/ChatRoom/ChatRoom.tsx [hunk 3]
```diff
@@ -399,14 +465,49 @@
...
+                </div>
+
+                <Button
+                  onClick={() => setReplyToDirectMessage(null)}
+                  className={styles.closeBtn}
+                  data-testid="closeReplyMsg"
+                >
+                  <Close />
+                </Button>
+              </div>
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/components/UserPortal/ChatRoom/ChatRoom.tsx:510]: Added test cases for reply message functionality.

File: src/components/UserPortal/ChatRoom/ChatRoom.tsx [hunk 4]
```diff
@@ -202,24 +211,17 @@
...
-        updatedChat?.messages.push(
-          directMessageSubscriptionData?.data.data.messageSentToDirectChat,
-        );
-        setDirectChat(updatedChat);
         chatRefetch();
+      } else {
+        chatRefetch({
+          id: directMessageSubscriptionData?.data.data.messageSentToDirectChat
+            .directChatMessageBelongsTo['_id'],
+        });
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/components/UserPortal/ChatRoom/ChatRoom.tsx:225]: Added tests for subscription handling

**Comment by palisadoes** [comment]
@DMills27 @Kevoniat Please review

**Review by DMills27 — APPROVED** [review]
LGTM

**Comment by DMills27** [comment]
@disha1202 Please resolve the conflicting files.

**Pull request merged**